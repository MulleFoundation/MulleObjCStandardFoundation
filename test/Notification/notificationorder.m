#ifndef __MULLE_OBJC__
# import <Foundation/Foundation.h>
#else
# import <MulleObjCStandardFoundation/MulleObjCStandardFoundation.h>
#endif


@implementation A : NSObject

+ (void) notification:(NSNotification *) notification
{
   printf( "A\n");
}

@end


@implementation B : NSObject

+ (void) notification:(NSNotification *) notification
{
   printf( "B\n");
}

@end


@implementation C : NSObject

+ (void) notification:(NSNotification *) notification
{
   printf( "C\n");
}

@end



void   test_abc( void)
{
   NSNotificationCenter   *center;

   printf( "%s\n", __FUNCTION__);

   @autoreleasepool
   {
      // simple leak checker
      center = [NSNotificationCenter new];
      [center addObserver:[C class]
                 selector:@selector( notification:)
                     name:@"Any"
                   object:nil];
      [center addObserver:[B class]
                 selector:@selector( notification:)
                     name:@"Any"
                   object:nil];
      [center addObserver:[A class]
                 selector:@selector( notification:)
                     name:@"Any"
                   object:nil];
      [center postNotificationName:@"Any"
                            object:nil];
      [center release];
   }
}


void   test_cba( void)
{
   NSNotificationCenter   *center;

   printf( "%s\n", __FUNCTION__);

   @autoreleasepool
   {
      // simple leak checker
      center = [NSNotificationCenter new];
      [center addObserver:[A class]
                 selector:@selector( notification:)
                     name:@"Any"
                   object:nil];
      [center addObserver:[B class]
                 selector:@selector( notification:)
                     name:@"Any"
                   object:nil];
      [center addObserver:[C class]
                 selector:@selector( notification:)
                     name:@"Any"
                   object:nil];
      [center postNotificationName:@"Any"
                            object:nil];
      [center release];
   }
}



int   main( void)
{
   NSNotificationCenter   *center;

   test_abc();
   test_cba();
   return( 0);
}
