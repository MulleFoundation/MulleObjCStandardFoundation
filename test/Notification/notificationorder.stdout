test_abc
A
B
C
test_cba
C
B
A
